subprojects/normalization-java/src/test/groovy/org/gradle/internal/normalization/java/ApiClassExtractorTestSupport.groovy

```diff
@@ -63,33 +64,33 @@ class ApiClassExtractorTestSupport extends Specification {
     }
 
     @CompileStatic
-    public static class ApiContainer {
+    static class ApiContainer {
         private final ApiClassLoader apiClassLoader = new ApiClassLoader()
         private final ApiClassExtractor apiClassExtractor
 
         public final Map<String, GeneratedClass> classes
 
-        public ApiContainer(List<String> packages, Map<String, GeneratedClass> classes) {
+        ApiContainer(List<String> packages, Map<String, GeneratedClass> classes) {
             this.apiClassExtractor = new ApiClassExtractor(packages.toSet())
             this.classes = classes
         }
 
         protected Class<?> extractAndLoadApiClassFrom(GeneratedClass clazz) {
-            apiClassLoader.loadClassFromBytes(apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes)))
+            apiClassLoader.loadClassFromBytes(apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes)).get())
         }
 
         protected byte[] extractApiClassFrom(GeneratedClass clazz) {
-            apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes))
+            apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes)).get()
         }
 
-        protected boolean shouldExtractApiClassFrom(GeneratedClass clazz) {
-            apiClassExtractor.shouldExtractApiClassFrom(new ClassReader(clazz.bytes))
+        protected boolean isApiClassExtractedFrom(GeneratedClass clazz) {
```

Comment:
I wonder if that method should be called `isNoApiClass` or something similar, given that we only use the `!` variant.

Reply:
We do use the direct variant in one place, too. I think the code is understandable either way, and is already better than what we had. I think we shouldn't try to improve it further for now, especially given that it's not some code we are actively working with anyway.

Reply:
Fair enough.